---
title: "Microbiome Section Report"
author: "Janelle"
date: "2024-04-18"
output: pdf_document
---
# Load required packages
```{r}
library(dada2)
```
# Load sequences
```{r}
path <- "sequences"
list.files(path)
```
# Read in file names
```{r}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```
# Inspect read quality
```{r}
plotQualityProfile(fnFs[1:2])
```
```{r}
plotQualityProfile(fnRs[1:2])
```
```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

# FReads decrease in quality after 150, RReads decreases in quality after 120
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(150,120),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE
head(out)
```
# Learn error rates for Forward Reads
```{r}
errF <- learnErrors(filtFs, multithread=FALSE)
```
# Learn error rates for Reverse Reads
```{r}
errR <- learnErrors(filtRs, multithread=FALSE)
```
# Visualization of error rates 
```{r}
plotErrors(errF, nominalQ=TRUE)
```
# Sample Inference of Forward Reads
```{r}
dadaFs <- dada(filtFs, err=errF, multithread=FALSE)
```
# Sample Inference of Reverse Reads
```{r}
dadaRs <- dada(filtRs, err=errR, multithread=FALSE)
```
# create sequence table
```{r}
seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)

# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
```
# Remove chimeras
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=FALSE, verbose=TRUE)
dim(seqtab.nochim)
```
# track reads through pipeline
```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), rowSums(seqtab.nochim))
```
# Save seqtab.nochim as an R file
```{r}
save(seqtab.nochim, file="RData/seqtab.nochim,RData")
```
# Load seqtab.nochim to start here
```{r}
load("RData/seqtab.nochim,RData")
```
# Assign Taxonomy
```{r}
# download the Silva species database from https://zenodo.org/records/4587955
taxa <- assignTaxonomy(seqtab.nochim, "silva_nr99_v138.1_wSpecies_train_set.fa.gz")
```
# save taxonomy
```{r}
save(taxa, file = "RData/taxa.RData")
```

